impl/client/src/main/javascript/web/dojo/pentaho/common/FieldList.js

```diff
@@ -911,16 +922,22 @@ define(["dojo/_base/declare", "dijit/_WidgetBase", "dijit/_Templated", "dojo/on"
     if (this._isHiddenChild(prevField)) {
       return this._findPrevField(prevField);
     }
+      var calButton = dom.byId("catId-add-button");
+      var tabIndexValue = this.isCalcField(prevField.id) ? "0" : "-1";
+      calButton.setAttribute("tabindex", tabIndexValue);
 
     return prevField;
   },
 
   _onKeydownHeader: function (e) {
-    var code = e.keyCode || e.which;
-    var header = e.target;
-    var categoryId = dojo.attr(header, "id");
-    var children;
-    var expanded = dojo.attr(header.firstElementChild, "collapsed") !== "true";
+      var code = e.keyCode || e.which;
+      var header = e.target;
+      var categoryId = dojo.attr(header, "id");
+      var children;
+      if (header.firstElementChild !== null){
+      var expanded = dojo.attr(header.firstElementChild, "collapsed") !== "true";
+      }
```

Comment:
Hi, @SreejaSridhar. In the future, please, let me (the reviewer) close the comments that I create — it's easier for the reviewer to know what he still needs to validate. I do ask you (the author) to mark the comment as addressed, somehow (e.g. by simply liking it, or adding "Addressed" as a comment), so that I (the reviewer) know it is ready for review. It is also the reviewer that can best tell if the ask was actually addressed, and he should close resolve the comment once checked in that case.

I believe this works well and is fairly common practice when working on the Pentaho product.
Thank you!